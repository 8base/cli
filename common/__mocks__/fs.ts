const fs = require("memfs");
fs.copyFileSync = (source, dest) => {
    fs.writeFileSync(dest, fs.readFileSync(source));
};
module.exports  = fs;