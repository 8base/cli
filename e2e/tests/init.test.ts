import * as fs from 'fs';
import * as execa from 'execa';

import { runner } from '../runner';

const TEST_DIR = 'testdir_init';

it('As a user, I can use help flag for see help information for `init` command.', async () => {
  const { stdout } = await runner()('init', '--help');

  expect(stdout).toMatchSnapshot();
});

it('As a user, I can use help flag for see help information for `init` command.', async () => {
  const { stdout } = await runner()('init', TEST_DIR);

  expect(stdout.replace(/Time: \d+ ms\./, 'Time: 100 ms.')).toMatchSnapshot();

  const { stdout: ls } = await execa.shell(`ls -1 ${TEST_DIR}/** | sort`);

  expect(ls.split("\n").sort()).toMatchSnapshot();
});

afterEach(async () => {
  await execa.shell(`rm -rf ${TEST_DIR}`);
});