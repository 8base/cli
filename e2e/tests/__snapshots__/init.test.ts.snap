// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As a user, I can use help flag for see help information for \`init\` command. 1`] = `
"8base init [OPTIONS]

Examples:
  8base init             initialize current folder
  8base init my_project  create folder my_project and initialize"
`;

exports[`As a user, I can use help flag for see help information for \`init\` command. 2`] = `
"Project testdir_init initialize success
init done. Time: 100 ms."
`;

exports[`As a user, I can use help flag for see help information for \`init\` command. 3`] = `
Array [
  "",
  "hello.graphql",
  "hello.ts",
  "hello2.graphql",
  "hello2.js",
  "testdir_init/8base.yml",
  "testdir_init/package.json",
  "testdir_init/src:",
]
`;
